### PR TITLE
feature(kubernetes): add kubernetes deployment

### PR DESCRIPTION
**Pull request by DanielHabenicht**
- Phonebook can be deployed in a kubernetes cluster now

### CODE REVIEW

**Comment by Zeroks77** [comment]
@DanielHabenicht du musst @paule96 schon als Reviewer hinzufügen :)

**Comment by paule96** [comment]
we should put all the kubernetes `.yml` in a seperate folder. So we know what `.yml` is for wich system.

**Comment by DanielHabenicht** [comment]
> we should put all the kubernetes `.yml` in a separate folder. So we know what `.yml` is for wich system.

Can you elaborate on what you mean?

**Comment by paule96** [comment]
we have  `.yml` for DevOps and for kubernetes and maybe for git versioning and so on... 

That must be maybe also done for the DevOps `.yml`. It's currently maybe a little bit hard to find the right `.yml`

**Comment by DanielHabenicht** [comment]
@paule96 please have a look again :)

File: Phonebook.Demo/Phonebook.Assets.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  selector:
+    matchLabels:
+      app: phonebook-assets
+  template:
+    metadata:
+      labels:
+        app: phonebook-assets
+    spec:
+      nodeSelector:
+        'beta.kubernetes.io/os': linux
```
Comment (paule96) [inline Phonebook.Demo/Phonebook.Assets.yml]: beta is always best practice 👍

File: Phonebook.Demo/Phonebook.Assets.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      containers:
+        - name: phonebook-assets
+          image: tsystemsmms/phonebook-build:assets-200
+          imagePullPolicy: 'Always'
+          resources:
+            requests:
+              cpu: 100m
```
Comment (paule96) [inline Phonebook.Demo/Phonebook.Assets.yml]: what means the 100m? Is that 1/100 Cpu?

File: Phonebook.Demo/Phonebook.MockBackend.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: phonebook-mock-backend-service
+spec:
+  ports:
+    - name: http
+      targetPort: 80
+      port: 80
```
Comment (paule96) [inline Phonebook.Demo/Phonebook.MockBackend.yml]: what happens if a port is used twice?

File: Phonebook.Demo/readme.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# Demo Application for the Phonebook
+
+## Prerequisites
+
+### Working Kubernetes Cluster
+
+#### Setup in Azure
+
+1. `az group create --subscription <id> --name phonebook-kubernetes --location northeurope`
```
Comment (paule96) [inline Phonebook.Demo/readme.md]:
```suggestion
> Note: To get the subscription id run `az account list`

1. `az group create --subscription <id> --name phonebook-kubernetes --location northeurope`
```

File: Phonebook.Source.MockPeopleBackend/index.js [hunk 5]
```diff
@@ -29,6 +29,7 @@ let longOrgUnits3 = longOrgUnits.slice(31, 50);
 let cities = generateArray(15, faker.address.city);
 
 let addresses = generateArray(30, faker.address.streetAddress);
+let roomNumbers = ['1A', '2A', '3A', '4A', '5A', '6A', '1B', '2B', '3B', '4B', '5B', '6B'];
```
Comment (paule96) [inline Phonebook.Source.MockPeopleBackend/index.js]: please revert the changes in this file. They are not relevant for the kubernetes deployment

File: Phonebook.Source.MockPeopleBackend/mocks/synthetic/branches.json [hunk 6]
```diff
@@ -1 +1 @@
...
..."Id":"1A","Number":"3A","Phone":"","Place":"Aldaview"},{"Building":"2099 Grayce Haven","BuildingId":31,"Description":"Aldaview, 96174 Towne Trail, Raum 3B","Floor":"4","FloorPlan":"sample","Id":"2A","Number":"6A","Phone":"","Place":"Aldaview"},{"Building":"56106 Garett Cape","BuildingId":29,"Description":"Aldaview, 6509 Crona Canyon, Raum 4B","Floor":"1","FloorPlan":"sample","Id":"5B","Number":"5B","Phone":"","Place":"Aldaview"},{"Building":"9551 Jacobs Stravenue","BuildingId":99,"Description":"Aldaview, 8124 Torp Ways, Raum 1B","Floor":"2","FloorPlan":"sample","Id":"6B","Number":"3B","Phone":"","Place":"Aldaview"},{"Building":"449 Katarina Bridge","BuildingId":68,"Description":"Aldaview, 8733 Alfonso Port, Raum 3B","Floor":"4","FloorPlan":"sample","Id":"4A","Number":"4A","Phone":"","Place":"Aldaview"},{"Building":"96174 Towne Trail","BuildingId":84,"Description":"Aldaview, 8101 Molly Point, Raum 2A","Floor":"0","FloorPlan":"sample","Id":"4B","Number":"4A","Phone":"","Place":"Aldaview"},{"Building":"0423 Leila Heights","BuildingId":51,"Description":"Aldaview, 6509 Crona Canyon, Raum 6A","Floor":"4","FloorPlan":"sample","Id":"1A","Number":"6A","Phone":"","Place":"Aldaview"},{"Buildi
```
Comment (paule96) [inline Phonebook.Source.MockPeopleBackend/mocks/synthetic/branches.json]: please revert the changes in this file. They are not relevant for the kubernetes deployment

File: Phonebook.Source.MockPeopleBackend/mocks/synthetic/rooms.json [hunk 7]
```diff
@@ -1 +1 @@
...
...17 Alverta Rest","BuildingId":100,"Description":"Kleinland, 95740 Deanna Cape, Raum 2A","Floor":"0","FloorPlan":"sample","Id":"4A","Number":"2A","Phone":"","Place":"Kleinland"},{"Building":"264 Wyman Tunnel","BuildingId":91,"Description":"Ziememouth, 9551 Jacobs Stravenue, Raum 5B","Floor":"2","FloorPlan":"sample","Id":"6B","Number":"5B","Phone":"","Place":"Ziememouth"},{"Building":"5340 Davis Grove","BuildingId":46,"Description":"North Emmittborough, 117 Alverta Rest, Raum 6B","Floor":"1","FloorPlan":"sample","Id":"5B","Number":"1B","Phone":"","Place":"North Emmittborough"},{"Building":"3902 McCullough Oval","BuildingId":80,"Description":"East Liaport, 79311 Streich Motorway, Raum 4A","Floor":"6","FloorPlan":"sample","Id":"4A","Number":"3B","Phone":"","Place":"East Liaport"},{"Building":"449 Katarina Bridge","BuildingId":68,"Description":"Aldaview, 8733 Alfonso Port, Raum 3B","Floor":"4","FloorPlan":"sample","Id":"4A","Number":"4A","Phone":"","Place":"Aldaview"},{"Building":"97791 Nikolaus Gateway","BuildingId":83,"Description":"Jaidafort, 97791 Nikolaus Gateway, Raum 4B","Floor":"2","FloorPlan":"sample","Id":"2A","Number":"6B","Phone":"","Place":"Jaidafort"},{"Building":"200 East
```
Comment (paule96) [inline Phonebook.Source.MockPeopleBackend/mocks/synthetic/rooms.json]: please revert the changes in this file. They are not relevant for the kubernetes deployment

File: Phonebook.Source.MockPeopleBackend/package-lock.json [hunk 8]
```diff
@@ -4,26 +4,10 @@
   "lockfileVersion": 1,
   "requires": true,
   "dependencies": {
```
Comment (paule96) [inline Phonebook.Source.MockPeopleBackend/package-lock.json]: please revert the changes in this file. They are not relevant for the kubernetes deployment

File: Phonebook/phonebook/templates/NOTES.txt [hunk 9]
```diff
@@ -0,0 +1,21 @@
+1. Get the application URL by running these commands:
+{{- if .Values.ingress.enabled }}
```
Comment (paule96) [inline Phonebook/phonebook/templates/NOTES.txt]: what for commands? I only see brackets....

File: Phonebook/phonebook/values.yaml [hunk 10]
```diff
@@ -0,0 +1,82 @@
+# Default values for phonebook.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+# General Variables
+host: &host 'demo-phonebook.me'
+contactUrl: 'https://github.com/T-Systems-MMS/phonebook/issues'
+roomPlanningToolUrl: 'undefined'
+contactEmail: &contactEmail 'daniel-habenicht@outlook.de'
```
Comment (paule96) [inline Phonebook/phonebook/values.yaml]: maybe we want create an issue for that and think about what e-mail address we should use for that. maybe we create a function mailbox so we all get information for something that went wrong there.

File: Phonebook/phonebook/values.yaml [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+employeePictureEndpoint: 'undefined'
+assetsEndpoint: 'https://demo-phonebook.me/assets'
+
+# Frontend Variables
+frontend:
+  name: frontend
+  image:
+    repository: tsystemsmms
+    name: phonebook-frontend
+    tag: 1.22.0
```
Comment (paule96) [inline Phonebook/phonebook/values.yaml]: how we change this tag? Can that be changed by an automate?

File: Phonebook/phonebook/values.yaml [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+    domain: board.demo-phonebook.me
+  ssl:
+    enabled: true
+    enforced: true
+  acme:
+    staging: false
+    enabled: true
+    email: *contactEmail
+    onHostRule: true
+    challengeType: 'http-01'
```
Comment (paule96) [inline Phonebook/phonebook/values.yaml]: is this the http specification we support? maybe we want support http2 also?

File: Phonebook.Demo/Phonebook.Assets.yml (hunk 1 above)
Comment (DanielHabenicht) [reply Phonebook.Demo/Phonebook.Assets.yml]: bleeding edge

File: Phonebook.Demo/Phonebook.MockBackend.yml (hunk 3 above)
Comment (DanielHabenicht) [reply Phonebook.Demo/Phonebook.MockBackend.yml]: idk, never tried it :)

File: Phonebook/phonebook/values.yaml (hunk 10 above)
Comment (DanielHabenicht) [reply Phonebook/phonebook/values.yaml]: good idea, your turn?

File: Phonebook/phonebook/values.yaml (hunk 12 above)
Comment (DanielHabenicht) [reply Phonebook/phonebook/values.yaml]: nope. That is just the challenge format for traefik.

**Comment by DanielHabenicht** [comment]
[ ] Write some docu
[ ] Split values.yml into general and demo part

File: Phonebook/phonebook/values.yaml (hunk 10 above)
Comment (paule96) [reply Phonebook/phonebook/values.yaml]:
```suggestion
contactEmail: &contactEmail 'phonebook-t-systems-mms@mg.telekom.de'
```

Comment (DanielHabenicht) [reply Phonebook/phonebook/values.yaml]: updated

Comment (mschwrdtnr) [reply Phonebook/phonebook/values.yaml]: i will also add this to docs later

**Comment by DanielHabenicht** [comment]
Yep, thanks just move the file than. Forgot to move it.
We also have to talk about how to handle this in the future it would be good to have a readme in each project folder.

Cheers
Daniel

________________________________

**Comment by mschwrdtnr** [comment]
@DanielHabenicht I meant the email adress..^^
